public/app/features/commandPalette/CommandPalette.tsx

```diff
@@ -66,14 +77,20 @@ export const CommandPalette = () => {
   ) : null;
 };
 
-const RenderResults = () => {
+interface RenderResultsProps {
+  dashboardResults: ActionImpl[];
+}
+
+const RenderResults = ({ dashboardResults }: RenderResultsProps) => {
   const { results, rootActionId } = useMatches();
   const styles = useStyles2(getSearchStyles);
+  const dashboardsSectionTitle = t('command-palette.section.dashboard-search-results', 'Dashboards');
+  const dashboards = dashboardResults.length > 0 ? [dashboardsSectionTitle, ...dashboardResults] : [];
 
   return (
     <div className={styles.resultsContainer}>
       <KBarResults
-        items={results}
+        items={[...results, ...dashboards]}
```

Comment:
useMemo, can probably happen in the same as the above

Reply:
I don't know whether it's definitely happening here, but changing the identity of an array that's rendered using a virtualised list can cause it to reset scroll position on every render, which is less than ideal!